pkg/oauth/facebook/facebook.go

```diff
@@ -0,0 +1,70 @@
+package facebook
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/alexandrevilain/postgrest-auth/pkg/config"
+	"github.com/alexandrevilain/postgrest-auth/pkg/model"
+	"github.com/alexandrevilain/postgrest-auth/pkg/oauth"
+)
+
+type facebookProvider struct {
+}
+
+//Facebookuser struct of facebook user
+type Facebookuser struct {
+	ID            string `json:"id"`
+	Email         string `json:"email"`
+	VerifiedEmail bool   `json:"verified_email"`
+	Name          string `json:"name"`
+	GivenName     string `json:"given_name"`
+	FamilyName    string `json:"family_name"`
+	Link          string `json:"link"`
+	Picture       string `json:"picture"`
+}
+
+// New retrive facebook user info based on the token provided
+func New() oauth.Provider {
+	return &facebookProvider{}
+}
+
+// GetUserInfo retrive facebook user info based on the token provided
+func (provider *facebookProvider) GetUserInfo(payload *oauth.Oauth2Payload, oauthStateString string, config *config.Config, db *sql.DB) (model.User, string, error) {
+	var facebookUser Facebookuser
+	if payload.State != oauthStateString {
+		return model.User{}, "", fmt.Errorf("invalid oauth state")
```

Comment:
Instead of returning model.User{} for each error, can you create a `var user model.User` at the beginning of the function? 
I think this is easier to read.